File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cs
public class Solution {
    public int FindMin(int[] nums) {
        int left = 0, right = nums.Length-1;
        int mid;

        while (left <= right) {
            if (nums[left] <= nums[right]) {
                return nums[left];
            }

            mid = (left + right)/2;

            if (nums[mid] >= nums[left]) {
                left = mid + 1;
            }
            else {
                right = mid;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []

        levels = {}
        level = 0
        q = [[root, 0]]

        while q:
            front, level = q.pop(0)

            if not level in levels:
                levels[level] = []
            levels[level].append(front.val)

            if front.left:
                q.append([front.left, 1 + level])
            if front.right:
                q.append([front.right, 1 + level])
        
        result = []

        for key, value in levels.items():
            result.append(value)
        
        return result

